solo-io/gloo
# gateway2: enable provisioning of AI extension

# Description
Enables the provisioning of the AI extproc extension via GatewayParameter. A new `aiExtension` field is added to enable customizing the deployment of the extension service as a sidecar. The extension service is opaque to Gloo OSS, and is left to downstream projects to implement and plug-in.

## API changes
- New field is added to GatewayParameter API

## Testing steps
Apply the following and confirm the extension sidecar is deployed:
```
k apply -f - <<EOF
apiVersion: gateway.gloo.solo.io/v1alpha1
kind: GatewayParameters
metadata:
  name: test
spec:
  kube:
    aiExtension:
      enabled: true
      listenAddress: <addr>
      image:
        registry: <registry>
        repository: <repo>
        tag: <tag>
        pullPolicy: IfNotPresent
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: test
  annotations:
    gateway.gloo.solo.io/gateway-parameters-name: "test"
spec:
  gatewayClassName: gloo-gateway
  listeners:
    - protocol: HTTP
      port: 8080
      name: http
      allowedRoutes:
        namespaces:
          from: All
---
EOF
```

## Notes for reviewers

Be sure to verify intended behavior by ...

Please proofread comments on ...

This is a complex PR and may require a huddle to discuss ...

# Checklist:
